crates/nostr-database/src/events/helper.rs

```diff
@@ -316,6 +317,14 @@ impl InternalDatabaseHelper {
                     to_discard.extend(self.internal_query_by_kind_and_author(params).map(|e| e.id));
                 }
             }
+        } else if kind == Kind::RequestToVanish {
+            // Remove all author events and gift wraps they p-tagged in it
+            to_discard.extend(self.events.iter().filter_map(|ev| {
+                (ev.pubkey == author
+                    || (ev.kind == Kind::GiftWrap
+                        && ev.tags.iter().any(|t| t == &Tag::public_key(author))))
```

Comment:
Here you can use this instead:

```rust
tags.public_keys().any(|p| p == &author)
```

In the above condition (`kind == Kind::EventDeletion`), any vanish request deletion should be ignored.
And the database should keep track of the vanish request and avoid that all events before the vanish request will not be stored again, like I've done for the coordinate deletion.


Also, I think something must be reworked in the database traits, because if a user want to delete the events only from a specific relay, this will delete all events also locally.
For now, if the vanish request contains the `ALL_RELAYS` tag, the request can be processed, otherwise should be ignored.

These things apply also to the other databases